filament/backend/src/metal/MetalHandles.mm

```diff
@@ -433,46 +433,38 @@ BufferDescriptor b(nullptr, 0u, [](void* buffer, size_t size, void* user) {
 
     const BOOL mipmapped = levels > 1;
     const BOOL multisampled = samples > 1;
-    const BOOL textureArray = target == SamplerType::SAMPLER_2D_ARRAY;
 
 #if defined(IOS)
     ASSERT_PRECONDITION(!textureArray || !multisampled,
             "iOS does not support multisampled texture arrays.");
 #endif
 
-    const auto get2DTextureType = [](bool isArray, bool isMultisampled) {
-        uint8_t value = 0;
-        if (isMultisampled) {
-            value |= 0b10u;
-        }
-        if (isArray) {
-            value |= 0b01u;
-        }
-        switch (value) {
-            default:
-            case 0b00:
-                return MTLTextureType2D;
-            case 0b01:
-                return MTLTextureType2DArray;
-            case 0b10:
-                return MTLTextureType2DMultisample;
-            case 0b11:
+    const auto get2DTextureType = [](SamplerType target, bool isMultisampled) {
+        if (!isMultisampled) {
+            return getMetalType(target);
+        } else {
+            switch (target) {
+                case SamplerType::SAMPLER_2D:
+                    return MTLTextureType2DMultisample;
 #if !defined(IOS)
-                return MTLTextureType2DMultisampleArray;
-#else
-                // should not get here
-                return MTLTextureType2DArray;
+                case SamplerType::SAMPLER_2D_ARRAY:
+                    return MTLTextureType2DMultisampleArray;
 #endif
+                default:
+                    // should not get here
+                    return getMetalType(target);
+            }
         }
     };
 
     MTLTextureDescriptor* descriptor;
     switch (target) {
         case SamplerType::SAMPLER_2D:
         case SamplerType::SAMPLER_2D_ARRAY:
+        case SamplerType::SAMPLER_CUBEMAP_ARRAY:
```

Comment:
I think we can put this `case` statement next to the existing `case SamplerType::SAMPLER_CUBEMAP` on line 478 by copying down the `descriptor.arrayLength = depth` line.

Reply:
done -- and also I think my version didn't work because the texture type was wrong.